jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/actions/CodeWhispererRecommendationAction.kt

```diff
@@ -19,11 +20,14 @@ class CodeWhispererRecommendationAction : AnAction(message("codewhisperer.trigge
     }
 
     override fun actionPerformed(e: AnActionEvent) {
+        val latencyContext = LatencyContext()
+        latencyContext.codewhispererPreprocessingStart = System.nanoTime()
+        latencyContext.codewhispererEndToEndStart = System.nanoTime()
```

Comment:
nit: maybe we can put it in the constructor?

Reply:
yea we can, slightly preferred not to personally due to the fact that constructing this doesn't necessarily bind to the start time of these two. It just happens to be this case.